activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -147,7 +151,7 @@ def self.define_validations(model, reflection)
       super
 
       if required
-        model.validates_presence_of reflection.name, message: :required
+        model.validates_presence_of reflection.name, message: (validation_message.presence || :required)
```

Comment:
I know, at most one validation is defined for `belongs_to` (and `has_one`) relation, and that is of presence only, so, this code will work. But the option you specified to be able to define a custom validation message i.e. `:message` does not correctly represent what validation is this message for, e.g. in case there are more than one validations possible (i can't think of any now, but future is uncertain). Just a point from me.

Reply:
May be `belongs_to :user, message: { presence: 'Custom error message' }` or something better.

Reply:
I believe the `:presence` key is not required here because the only validation implied now is `required`. So here it is intended to do.

```
   belongs_to :user, message: 'must be present.'
```